### PR TITLE
Update our sorting/filtering pattern

### PR DESCRIPTION
**Pull request by andrewmcodes**
# Enhancement

## Description

- Establish clearer pattern for sorting/filtering
- Update SR
- Update pagy and fix associated TODO's

## Why should this be added

Creates a much clearer pattern that is also faster.

### General

- [x] My code follows the style guidelines of this project & the tests are passing
- [x] I have performed a self-review of my own code
- [x] I have updated DB schema annotations or I have not made any DB schema changes

### CodeFund Core Team

- [x] I have added the appropriate milestone if applicable
- [x] I have requested a review from another developer(s)

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -17,10 +17,17 @@ class ApplicationController < ActionController::Base
...
   impersonates :user
 
   protected
 
+  def set_last_controller_name
+    unless controller_name.include? "sparkline"
+      session[:last_controller_name] = controller_name
+    end
+  end
+
```
Comment (hopsoft) [inline app/controllers/application_controller.rb]: Are we still using `last_controller_name` anywhere?

File: app/controllers/concerns/scopable.rb [hunk 2]
```diff
@@ -2,26 +2,16 @@ module Scopable
...
 
   included do
-    helper_method :scope_list
-    before_action :reset_current_scope
+    before_action :set_scoped_by
   end
 
   def scope_list(model)
-    model.public_send(current_scope)
+    model.public_send @scoped_by
```
Comment (hopsoft) [inline app/controllers/concerns/scopable.rb]: We should probably add a TODO to tighten up security on this.

File: app/javascript/controllers/index.js [hunk 3]
```diff
@@ -4,8 +4,9 @@
 import { Application } from 'stimulus'
 import { definitionsFromContext } from 'stimulus/webpack-helpers'
 import StimulusReflex from 'stimulus_reflex'
+import consumer from '../channels/consumer'
 
 const application = Application.start()
 const context = require.context('controllers', true, /_controller\.js$/)
 application.load(definitionsFromContext(context))
-StimulusReflex.initialize(application)
+StimulusReflex.initialize(application, { consumer })
```
Comment (hopsoft) [inline app/javascript/controllers/index.js]: Nice.

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (andrewmcodes) [reply app/controllers/application_controller.rb]: good catch

File: app/controllers/concerns/scopable.rb [hunk 4]
```diff
@@ -2,16 +2,22 @@ module Scopable
...
   protected
 
   def set_scoped_by
     @scoped_by ||= "all"
   end
+
+  # Abstract method that should be overridden in including controllers
+  def scopable_values
+    raise NotImplementedError, "controller should implement `scopable_values`"
+  end
```
Comment (hopsoft) [inline app/controllers/concerns/scopable.rb]: Shouldn't this method be named `set_scopable_values`?

**Pull request merged**